### PR TITLE
Don't request unsupported rates from providers #18081

### PR DESCRIPTION
**Pull request by martinvy**
https://is.roihunter.com/issues/18081

If a specific exchange rate miss from a provider, the provider is requested for the rate. But if the provider does not support rates for such currency, it will always fail and the process will be again repeated. We should find out what currencies are supported by which providers and we should avoid making of useless requests.

@business-factory/pythonistas please review & merge.

### CODE REVIEW

**Comment by Joozty** [comment]
@martinvy Nice solution with using cache. :)

Maybe you should add to logger some warning/error when you are returning empty set(). It seems at least suspiciously. You know URL can still exist but structure of document can change and your regular expression will no longer work.

And I think __timeout__ parameter in request.get() it's good to use. Server can throttling you and you will stuck there "forever". :smile: Yes, it's more unlikely but you never know.

**Comment by martinvy** [comment]
@Joozty Thanks for review, I updated the code according to your suggestions :)

File: gold_digger/data_providers/_provider.py [hunk 1]
```diff
@@ -25,23 +36,23 @@ def get_historical(self, origin_date, currencies):
...
                 return response
             else:
-                self.logger.error("%s get %s status code on %s" % (self, response.status_code, url))
+                self.logger.error("%s - status code: %s, URL: %s, Params: %s", self, response.status_code, url, params)
         except requests.exceptions.RequestException as e:
-            self.logger.error("%s get %s exception on %s" % (self, e, url))
+            self.logger.error("%s - exception: %s, URL: %s, Params: %s", self, e, url, params)
 
-    def _post(self, url, **kwargs):
+    def _post(self, url, params=None):
```
Comment (harpener) [inline gold_digger/data_providers/_provider.py]: @martinvy I checked that we only use POST for Yahoo and it returns the same results for GET. Is it needed somehow? :-)

Comment (martinvy) [reply gold_digger/data_providers/_provider.py]: @harpener I don't remember. But I would investigate it in the task for Yahoo.

**Comment by miso-belica** [comment]
Hope you are not waiting for me for review. I trust you guys, so feel free to merge :)

**Pull request merged**